src/link/federation_spec_definition.rs

```diff
@@ -3,22 +3,23 @@ use crate::link::spec::{Identity, Url, Version};
 use crate::link::spec_definition::{SpecDefinition, SpecDefinitions};
 use crate::schema::FederationSchema;
 use apollo_compiler::ast::Argument;
-use apollo_compiler::schema::{Directive, DirectiveDefinition, Value};
-use apollo_compiler::{Node, NodeStr};
+use apollo_compiler::schema::{Directive, DirectiveDefinition, Name, Value};
+use apollo_compiler::{name, Node, NodeStr};
 use lazy_static::lazy_static;
 
-pub(crate) const FEDERATION_KEY_DIRECTIVE_NAME_IN_SPEC: &str = "key";
-pub(crate) const FEDERATION_INTERFACEOBJECT_DIRECTIVE_NAME_IN_SPEC: &str = "interfaceObject";
-pub(crate) const FEDERATION_EXTERNAL_DIRECTIVE_NAME_IN_SPEC: &str = "external";
-pub(crate) const FEDERATION_REQUIRES_DIRECTIVE_NAME_IN_SPEC: &str = "requires";
-pub(crate) const FEDERATION_PROVIDES_DIRECTIVE_NAME_IN_SPEC: &str = "provides";
-pub(crate) const FEDERATION_SHAREABLE_DIRECTIVE_NAME_IN_SPEC: &str = "shareable";
-pub(crate) const FEDERATION_OVERRIDE_DIRECTIVE_NAME_IN_SPEC: &str = "override";
+pub(crate) static FEDERATION_KEY_DIRECTIVE_NAME_IN_SPEC: Name = name!("key");
```

Comment:
This and other `Name` constants can be `const` items, no need to make them `static`. (The difference is that statics have a fixed address in the memory-mapped executable and can only be accessed by `&` reference.)

Reply:
Chatted with Simon about this in DMs. My initial reasoning for not using `const` here was that I was worried that we would be allocating space for the strings in the `Name`s for each use. It turns out though that `name!()` uses `Name::from_static()`, which avoids the extra allocation and ref-counting for `&'static str`. Will change back to `const`s.